core-bundle/src/Security/Voter/CoreBundleVisibleElementVoter.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Security\Voter;
+
+use Contao\ArticleModel;
+use Contao\ContentModel;
+use Contao\FrontendUser;
+use Contao\ModuleModel;
+
+class CoreBundleVisibleElementVoter extends AbstractFrontendAccessVoter
+{
+    protected function supportsSubject($subject): bool
+    {
+        return $subject instanceof ContentModel
+            || $subject instanceof ModuleModel
+            || $subject instanceof ArticleModel;
```

Comment:
I think this is an unnecessary limitation. The visible element voter should be able to vote on any model that has an `->invisible` field (as well as a published voter should be able to vote on any model that has a `->published` field).

Reply:
We can't implement that. The voter would _automatically_ vote on any element without knowing if he's supposed to. The only option I can think of is **explicitly** ask the voter to, e.g. by calling the attribute `contao_frontend.is_visible_element`. But the current attribute is supposed to be _universal_ and have a different voter for different subjects.

Reply:
Using `contao_frontend.is_visible_element` would mean the caller would need to decide if the subject is a valid element for the voter. She would need to know if the subject implements `->protected` and/or `->guest` (`published` is not in the scope of this PR 😉). I don't think that would be correct, but honestly don't have much experience with voters, maybe @sheeep or @bytehead can tell?

Reply:
Imo even being a content model does not ensure these fields contain correct data: I could implement my own content element with this model and omit a field in the palette - then the values are 'dangling'. :shrug:

Reply:
that's correct, but I'd argue that's a bug in your code then to leave them away – because that's a default feature of every content element, and it (imho) never makes sense to remove them.

Reply:
Hmm, would this mean I have to go over all my custom content elements where I don't care about the `protected`/`groups`/`guests` properties and add those / add a load callback that sets `protected` to `''` to have a sane default?

If I understand correctly, the Voter wouldn't only be used by my business code for the custom content element but the also the system itself, right?

Reply:
No, I would argue that your elements should use these fields 😉 
Anyway, this PR does not change the existing behavior, it's exactly as-is just implemented in Symfony Security.